crates/filtering/src/madgwick.rs

```diff
@@ -0,0 +1,94 @@
+use nalgebra::{
+    Matrix4, Quaternion, RealField, Scalar, SimdValue, UnitQuaternion, Vector3, Vector4,
+};
+use num_traits::{One, Zero};
+
+#[derive(Debug)]
+pub struct AccelerometerNormZero;
+
+/// Madgwick's filter for orientation estimation.
+///
+/// This implementation is based on the following:
+/// https://github.com/jmagnuson/ahrs-rs
+pub trait Madgwick<T> {
+    fn update_with_imu(
+        &mut self,
+        gyroscope: &Vector3<T>,
+        accelerometer: &Vector3<T>,
+        filter_gain: T,
+        sample_period: T,
```

Comment:
That would also allow to have this properly typed as `Duration`

Reply:
yes